scripts/generate_black_spots_input.py

```diff
@@ -0,0 +1,374 @@
+#!/usr/bin/env python
+
+"""Loads records and roads shapefile, outputs data needed for generating black spots"""
+import argparse
+import csv
+from dateutil import parser
+from dateutil.relativedelta import relativedelta
+import fiona
+from functools import partial
+import itertools
+import logging
+import os
+import pyproj
+import pytz
+import rtree
+from shapely.geometry import mapping, shape, LineString, Point
+from shapely.ops import transform, unary_union
+
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger()
+
+
+def read_roads(roads_shp):
+    """Reads shapefile and extracts roads and projection"""
+    shp_file = fiona.open(roads_shp)
+    return ([shape(road['geometry']) for road in shp_file], shp_file.crs)
+
+
+def get_intersections(roads):
+    """Calculates the intersection points of all roads"""
+    intersections = []
+    for road1, road2 in itertools.combinations(roads, 2):
+        if road1.intersects(road2):
+            intersection = road1.intersection(road2)
+            if 'Point' == intersection.type:
+                intersections.append(intersection)
+            elif 'MultiPoint' == intersection.type:
+                intersections.extend([pt for pt in intersection])
+            elif 'MultiLineString' == intersection.type:
+                multiLine = [line for line in intersection]
+                first_coords = multiLine[0].coords[0]
```

Comment:
I'm not sure this handles the case where the two roads overlap in several places but it may not be worth worrying about since roads shouldn't really do that.